be/src/util/once.h

```diff
@@ -1,78 +1,54 @@
-// Licensed to the Apache Software Foundation (ASF) under one
-// or more contributor license agreements.  See the NOTICE file
-// distributed with this work for additional information
-// regarding copyright ownership.  The ASF licenses this file
-// to you under the Apache License, Version 2.0 (the
-// "License"); you may not use this file except in compliance
-// with the License.  You may obtain a copy of the License at
-//
-//   http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing,
-// software distributed under the License is distributed on an
-// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-// KIND, either express or implied.  See the License for the
-// specific language governing permissions and limitations
-// under the License.
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
 
 #pragma once
 
+#include <bthread/sys_futex.h>
+
 #include <atomic>
+#include <functional>
 
-#include "storage/olap_common.h"
+#include "common/statusor.h"
 
 namespace starrocks {
 
-// Utility class for implementing thread-safe call-once semantics.
-//
-// call() will return stored result regardless of whether the first invocation
-// returns a success status or not.
-//
-// Example:
-//   class Resource {
-//   public:
-//     Status init() {
-//       _init_once.call([this] { return _do_init(); });
-//     }
-//
-//     bool is_inited() const {
-//       return _init_once.has_called() && _init_once.stored_result().ok();
-//     }
-//   private:
-//     Status _do_init() { /* init logic here */ }
-//     StarRocksCallOnce<Status> _init_once;
-//   };
-template <typename ReturnType>
-class StarRocksCallOnce {
+class SuccessOnce {
```

Comment:
Could you add some comments for this class to explain why and how to use it.

Reply:
updated